test/testPriceRouter/PriceRouter.t.sol

```diff
@@ -0,0 +1,788 @@
+    }
+
+    function testAssetStalePrice() external {
+        // Store timestamp of USDC.
+        uint256 timestamp = uint256(IChainlinkAggregator(USDC_USD_FEED).latestTimestamp());
+        timestamp = block.timestamp - timestamp;
+
+        // Advance time so that the price becomes stale.
+        vm.warp(block.timestamp + 1 days);
+
+        vm.expectRevert(
+            bytes(
+                abi.encodeWithSelector(
+                    PriceRouter.PriceRouter__StalePrice.selector,
+                    address(USDC),
+                    timestamp + 1 days,
+                    1 days
+                )
+            )
+        );
+        priceRouter.getValue(USDC, 1e6, USDC);
+    }
+
+    function testETHtoUSDPriceFeedIsChecked() external {
+        // Check if querying an asset that needs the ETH to USD price feed, that the feed is checked.
+        // Add BOND as an asset.
+        PriceRouter.AssetSettings memory settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, BOND_ETH_FEED);
+        PriceRouter.ChainlinkDerivativeStorage memory stor = PriceRouter.ChainlinkDerivativeStorage(
+            100e18,
+            0.0001e18,
+            2 days,
+            true
+        );
+
+        // Simulate calling startEditAsset.
+        bytes32 editHash = keccak256(abi.encode(BOND, settings, abi.encode(stor)));
+        stdstore
+            .target(address(priceRouter))
+            .sig(priceRouter.assetEditableTimestamp.selector)
+            .with_key(editHash)
+            .checked_write(uint256(1));
+
+        priceRouter.completeEditAsset(BOND, settings, abi.encode(stor), 4.18e8);
+
+        // Re-add WETH, but shorten the heartbeat.
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, WETH_USD_FEED);
+        stor = PriceRouter.ChainlinkDerivativeStorage(0, 0.0, 3600, false);
+
+        // Simulate calling startEditAsset.
+        editHash = keccak256(abi.encode(WETH, settings, abi.encode(stor)));
+        stdstore
+            .target(address(priceRouter))
+            .sig(priceRouter.assetEditableTimestamp.selector)
+            .with_key(editHash)
+            .checked_write(uint256(1));
+
+        priceRouter.completeEditAsset(WETH, settings, abi.encode(stor), 1_775e8);
+
+        uint256 timestamp = uint256(IChainlinkAggregator(WETH_USD_FEED).latestTimestamp());
+        timestamp = block.timestamp - timestamp;
+
+        // Advance time forward such that the ETH USD price feed is stale, but the BOND ETH price feed is not.
+        vm.warp(block.timestamp + 3600);
+
+        vm.expectRevert(
+            bytes(
+                abi.encodeWithSelector(
+                    PriceRouter.PriceRouter__StalePrice.selector,
+                    address(WETH),
+                    timestamp + 3600,
+                    3600
+                )
+            )
+        );
+        priceRouter.getValue(BOND, 1e18, USDC);
+    }
+
+    function testAddingATwapAsset() external {
```

Comment:
Repeating my inquiry at the bottom of this test contract, but I think it would be worthwhile to test various derivative's exchange pricing.

Reply:
That would add more testing to the Twap asset setup and pricing

Reply:
So 95% of the pricing logic for TWAPs is coming from UniV3s oracle library, so checking that a TWAP asset price is returning something reasonable is sufficient to see if we set it up correctly.

Reply:
I added an RPL case to the exchangeRate test